### PR TITLE
[clang][Interp] Implement IntegralAP::{div, rem}

### PR DESCRIPTION
**Pull request by tbaederr**

### CODE REVIEW

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

File: clang/test/AST/Interp/intap.cpp [hunk 1]
```diff
@@ -44,6 +44,24 @@ static_assert(MulA * MulB == 50, ""); // ref-error {{not an integral constant ex
 static_assert(MulA * 5 == 25, "");
 static_assert(-1 * MulB == -7, "");
 
+
+constexpr _BitInt(4) DivA = 2;
+constexpr _BitInt(2) DivB = 1;
+static_assert(DivA / DivB == 2, "");
+
+constexpr _BitInt(4) DivC = DivA / 0; // ref-error {{must be initialized by a constant expression}} \
```
Comment (AaronBallman) [inline clang/test/AST/Interp/intap.cpp:52]: I'd appreciate another test along the lines of:
```
constexpr int bottom = -1;
constexpr int top = INT_MIN;
constexpr int nope = top / bottom;
constexpr int noooo = top % bottom;
```

**Review by AaronBallman — APPROVED** [review]
LGTM aside from a test request.

File: clang/test/AST/Interp/intap.cpp (hunk 1 above)
Comment (tbaederr) [reply clang/test/AST/Interp/intap.cpp:52]: Did you really mean `int`? Or `_BitInt`?

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

File: clang/test/AST/Interp/intap.cpp (hunk 1 above)
Comment (AaronBallman) [reply clang/test/AST/Interp/intap.cpp:52]: I meant `int`; basically, it's catching the other form of UB that's not division by zero.

Comment (tbaederr) [reply clang/test/AST/Interp/intap.cpp:53]: Using `int` means it won't use `IntegralAP` at all though. I've added those tests with `_BitInt(32)`.

Comment (AaronBallman) [reply clang/test/AST/Interp/intap.cpp:53]: Ah, thank you!

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Pull request merged**